src/os/unix/wayland.rs

```diff
@@ -0,0 +1,1117 @@
+                        let mut state = keymap.state();
+                        let mut update = state.update();
+                        update.mask(mods_depressed, mods_latched, mods_locked, 0, 0, group);
+                    }
+                }
+                _ => {}
+            }
+        }
+
+        self.scroll_x = 0.;
+        self.scroll_y = 0.;
+
+        for event in self.input.iter_pointer_events() {
+            use wayland_client::protocol::wl_pointer::Event;
+
+            match event {
+                Event::Enter {
+                    serial,
+                    surface_x,
+                    surface_y,
+                    ..
+                } => {
+                    self.mouse_x = surface_x as f32;
+                    self.mouse_y = surface_y as f32;
+                    self.input.get_pointer().set_cursor(
+                        serial,
+                        Some(&self.display.cursor_surface),
+                        0,
+                        0,
+                    );
+                    self.display
+                        .update_cursor(Self::decode_cursor(self.prev_cursor));
+                }
+                Event::Motion {
+                    surface_x,
+                    surface_y,
+                    ..
+                } => {
+                    self.mouse_x = surface_x as f32;
+                    self.mouse_y = surface_y as f32;
+                }
+                Event::Button { button, state, .. } => {
+                    use wayland_client::protocol::wl_pointer::ButtonState;
+
+                    let st = (state == ButtonState::Pressed) as u8;
+
+                    match button {
+                        //Left
+                        KEY_MOUSE_BTN1 => self.buttons[0] = st,
+                        //Right
+                        KEY_MOUSE_BTN2 => self.buttons[1] = st,
+                        //Middle
+                        KEY_MOUSE_BTN3 => self.buttons[2] = st,
+                        _ => {}
+                    }
+                }
+                Event::Axis { axis, value, .. } => {
+                    use wayland_client::protocol::wl_pointer::Axis;
+
+                    match axis {
+                        Axis::VerticalScroll => self.scroll_y = value as f32,
+                        Axis::HorizontalScroll => self.scroll_x = value as f32,
+                        _ => {}
+                    }
+                }
+                //Event::Frame {} => {}
+                //Event::AxisSource { axis_source } => {}
+                Event::AxisStop { axis, .. } => {
+                    use wayland_client::protocol::wl_pointer::Axis;
+
+                    match axis {
+                        Axis::VerticalScroll => self.scroll_y = 0.,
+                        Axis::HorizontalScroll => self.scroll_x = 0.,
+                        _ => {}
+                    }
+                }
+                //Event::AxisDiscrete { axis, discrete } => {}
```

Comment:
See my question above.

```suggestion
                // Event::AxisDiscrete { axis, discrete } => {}
```